pyo3-macros-backend/src/pyclass.rs

```diff
@@ -1637,6 +1666,32 @@ fn impl_pytypeinfo(
     }
 }
 
+fn pyclass_hash(
+    options: &PyClassPyO3Options,
+    cls: &syn::Type,
+    ctx: &Ctx,
+) -> Result<(Option<syn::ImplItemFn>, Option<MethodAndSlotDef>)> {
+    if options.hash.is_some() && options.frozen.is_none() {
+        bail_spanned!(options.hash.span() => "The `hash` option can only be using in conjunction with `frozen`.");
```

Comment:
Even though I personally like the phrase "in conjunction with", I think the following simplification might read easier to those with limited english:

```suggestion
        bail_spanned!(options.hash.span() => "The `hash` option requires the `frozen` option");
```